### PR TITLE
Merge upstream and add testcase

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: tests/blocks/sugarchain_mainnet_4257236.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "hash": "43d0eb06c959e39ddadfa3ddc13967143a203e01d318b9c3f960b817501dd2ee",
+  "size": 280,
+  "height": 1012,
```
Comment (decryp2kanon) [inline tests/blocks/sugarchain_mainnet_4257236.json]: its `Weight`. not height. `4257236`
https://1explorer.sugarchain.org/block/43d0eb06c959e39ddadfa3ddc13967143a203e01d318b9c3f960b817501dd2ee

File: tests/blocks/sugarchain_mainnet_4257236.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  "merkleroot": "ca7de6b83f3665810fee4e7e693ec68e731a67fbe6e07fe3c4ed071d41648ec5",
+  "tx": [
+    "ca7de6b83f3665810fee4e7e693ec68e731a67fbe6e07fe3c4ed071d41648ec5"
+  ],
+  "time": 1587900212,
+  "nonce": 768,
+  "bits": "1e00e46d",
+  "previousblockhash": "8757618e1dc6ff75d99ee3a10762b9b2c75a5c7ea24693d78989451fef54e4f6",
+  "block": "00000020f6e454ef1f458989d79346a27e5c5ac7b2b96207a1e39ed975ffc61d8e615787c58e64411d07edc4e37fe0e6fb671a738ec63e697e4eee0f8165363fb8e67dca346fa55e6de4001e0003000001010000000001010000000000000000000000000000000000000000000000000000000000000000ffffffff2203d4f54004346fa55e08580001b9000000000f706f6f6c2e72706c616e742e78797a00000000020000000000000000266a24aa21a9ede2f61c3f71d1defd3fa999dfa36953755c690689799962b48bebd836974e8cf900000000010000001600149c838f94cdda9c8fe15b07d68b8d70b194bdc50e0120000000000000000000000000000000000000000000000000000000000000000000000000"
```
Comment (decryp2kanon) [inline tests/blocks/sugarchain_mainnet_4257236.json:13]: where did you get this value?

File: tests/blocks/sugarchain_mainnet_4257236.json (hunk 1 above)
Comment (ghost) [reply tests/blocks/sugarchain_mainnet_4257236.json]: Nice catch, I have fixed them

File: tests/blocks/sugarchain_mainnet_4257236.json (hunk 2 above)
Comment (ghost) [reply tests/blocks/sugarchain_mainnet_4257236.json:13]: `sugarchain-cli getblock 43d0eb06c959e39ddadfa3ddc13967143a203e01d318b9c3f960b817501dd2ee 0`

Comment (decryp2kanon) [reply tests/blocks/sugarchain_mainnet_4257236.json:13]: i can confirm. thanks!

**Comment by decryp2kanon** [comment]
@volbil @y-chan help i am an electrum newbie :rofl:

**Review by y-chan — APPROVED** [review]
LGTM

**Comment by ghost** [comment]
Testnet support has been added

**Comment by volbil** [comment]
LGTM

**Review by volbil — APPROVED** [review]
Looking good so far

**Pull request closed without merging**